### ISSUE TITLE
About testing in SAMP

### ISSUE THREAD

**Issue by phj128** [mohamedhassanmus/SAMP#3]
Hi， thanks for your awesome work.
I have some questions about testing in SAMP. 
1. In table3, how do you calculate the excution time and verify whether it achieve or not?
2. In table4, SAMP is stochastic, do you test some times and average them?
3. In table5, are the results calculated from comparing outputs with the same inputs in test data?
 
I am also wondering whether you could provide the test code of SAMP? If this repo already contains them, please tell me where it is.
Thank you.

**Comment by mohamedhassanmus** [comment]
Hi, 
Thank you for your interest!

1 - Execution time is the time required to transition to the target action label from an idle state. The action label is a continuous vector between 0 and 1. So if `target_action > 0.9` we consider the action to have been executed and compute the execution time.
2. Yes we run the model multiple times and compute the average.
3.  I am not sure what does this question means.

**Comment by phj128** [comment]
Thank you for your answers.

For question3, I mean do you test with the groundtruth input of x_(t-1)(containing pose, traj, action and others), sample with standard normal distribution, predict y_t and calculate the error between the groundtruth output y_t?

Also for the results in table4, do you test with the groundtruth goal trajectory input and autoregress others or you only give the start point and end point, generate the whole sequence with A*? 

In table4, do you calculate the distance with normalized data or unnormalized data?

**Comment by mohamedhassanmus** [comment]
Hi, 

Ground truth data is only used for computing the FD score. For all other evaluations, no ground truth data is used.

For all experiments, we initialize the character from an idle pose. All subsequent frames are generated by our method, no ground truth data or features are used.

We only provide the end goal point, and the whole path is computed by our method.

Table 4 is computed on the raw features without normalization.

**Comment by phj128** [comment]
Hi,

I notice that in the provided test sequences, the objects are same as the training sequences. However, in your paper you mention that you do all the testing with unseen objects randomly selected from ShapeNet. **How do you obtain the sequence on the new object? Maybe modify the test sequence using IK?**

Another question is about the test on NSM dataset. **I did not find the test set in NSM dataset so do you split their dataset to train and test? May I know the way to split them?**

**Comment by nicolasugrinovic** [comment]
@mohamedhassanmus thanks for sharing your work! 

1. Can you point out **where** is the evaluation **code** for computing the **FD score**? 
2. Also if possible could you share **details** about the **FD eval**? For instance:

-   1. Do you use vPoser features for FD score computation? If so, I guess it is the latent space of 32-d?
-   2. Do you train a classifier on top of this to refine the features?
-   3. Do you evaluate each pose in a sequence independently?

**Comment by mohamedhassanmus** [comment]
> Hi,
> 
> I notice that in the provided test sequences, the objects are same as the training sequences. However, in your paper you mention that you do all the testing with unseen objects randomly selected from ShapeNet. **How do you obtain the sequence on the new object? Maybe modify the test sequence using IK?**
> 
> Another question is about the test on NSM dataset. **I did not find the test set in NSM dataset so do you split their dataset to train and test? May I know the way to split them?**

@phj128 Apologies for the late response.
All evaluations were done on unseen objects as mentioned. The test objects are in the main demo scene If I remembered correctly. The evaluation we reported in the paper (e.g. Table.5) does not require any test motion sequences. The metrics can be computed by testing the model directly on new objects in Unity in real-time.
For choosing the hyperparameters for the model, we chose test sequences. These were hardcoded in the Motion Editor and Exporter code in Unity.

We trained on the whole available NSM dataset and tested on new objects from ShapeNet

**Comment by mohamedhassanmus** [comment]
> @mohamedhassanmus thanks for sharing your work!
> 
> 1. Can you point out **where** is the evaluation **code** for computing the **FD score**?
> 2. Also if possible could you share **details** about the **FD eval**? For instance:
> 
> * 1. Do you use vPoser features for FD score computation? If so, I guess it is the latent space of 32-d?
> * 2. Do you train a classifier on top of this to refine the features?
> * 3. Do you evaluate each pose in a sequence independently?

The FD formula can be found in Wikipedia https://en.wikipedia.org/wiki/Fr%C3%A9chet_distance
under `As a distance between probability distributions (the FID score)`
or this paper https://www.sciencedirect.com/science/article/pii/0047259X8290077X

**Comment by nicolasugrinovic** [comment]
@mohamedhassanmus thanks for the links, very useful. 

Let me insist on the other points, sorry about that. As I understand the FID score has the I for inception, as usually the distances were measured using the feature space of an inception network trained as an image classifier. In the case of human poses, what features do you use? Can you provide some detail, I did not find it in the paper. Maybe I am missing something?

Thanks

**Comment by mohamedhassanmus** [comment]
@nicolasugrinovic We measure the Fréchet distance (FD) and not the FID score. Thus, there are no network features. The features used to compute the FD are specified in Eq.9 in the paper. The FD formula is in the link I mentioned above. So you can use the FD formula on the features from Eq.9 directly.

**Comment by nicolasugrinovic** [comment]
Thanks a lot
